activestorage/app/models/active_storage/blob/representable.rb

```diff
@@ -112,10 +112,10 @@ def default_variant_format
     end
 
     def format
-      if filename.extension.present? && MiniMime.lookup_by_extension(filename.extension)&.content_type == content_type
+      if filename.extension.present? && Marcel::MimeType.for(extension: filename.extension) == content_type
```

Comment:
Using fallback version here, if nothing matches it returns "application/octet-stream" which would fail the check